daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/scenario/ScenarioLedger.scala

```diff
@@ -257,12 +258,13 @@ object ScenarioLedger {
   }
 
   /** The initial ledger */
-  def initialLedger(t0: Time.Timestamp): ScenarioLedger =
+  def apply(
+      pkgSignature: Map[PackageId, language.Ast.GenPackage[_]],
+      currentTime: Time.Timestamp,
+  ): ScenarioLedger =
     ScenarioLedger(
-      currentTime = t0,
-      scenarioStepId = TransactionId(0),
-      scenarioSteps = immutable.IntMap.empty,
-      ledgerData = LedgerData.empty,
+      interfaceInstances = new language.util.PackageInfo(pkgSignature).instanceInterfaces,
```

Comment:
related to my other comment, it's confusing that here we assign `interfaceInstances = ...instanceInterfaces`. I'd rather call both `interfaceInstancesByTemplate`